tests/python/gaia-ui-tests/gaiatest/tests/functional/settings/test_settings_passcode.py

```diff
@@ -17,9 +17,12 @@ def test_set_passcode_by_settings(self):
         screen_lock_settings = settings.open_screen_lock_settings()
 
         screen_lock_settings.enable_passcode_lock()
+        old_passcode_code = self.data_layer.get_setting(
+            'lockscreen.passcode-lock.digest.value')
         screen_lock_settings.create_passcode(self._input_passcode)
 
         passcode_code = self.data_layer.get_setting('lockscreen.passcode-lock.code')
         passcode_enabled = self.data_layer.get_setting('lockscreen.passcode-lock.enabled')
-        self.assertEqual(passcode_code, "".join(self._input_passcode), 'Passcode is "%s", not "%s"' % (passcode_code, "".join(self._input_passcode)))
+        self.assertNotEqual(passcode_code, old_passcode_code, 'Passcode did not change.')
```

Comment:
I think it's probably okay to just duplicate the few lines for unlocking to homescreen. Something like:

```python
# lock screen and check new passcode
self.device.lock()
lock_screen = LockScreen(self.marionette)
lock_screen.switch_to_frame()
passcode_pad = lock_screen.unlock_to_passcode_pad()
homescreen = passcode_pad.type_passcode(self._input_passcode)
self.wait_for_condition(lambda m: self.apps.displayed_app.name == homescreen.name)
```

The above might fail if we unlock to the Settings app, so we might first want to close settings, or instead verify that the displayed app is Settings.